packages/css/src/util/fetch/PausableFetcher.ts

```diff
@@ -0,0 +1,55 @@
+import { getLoggerFor } from '@solid/community-server';
+import type { StatusDependant } from './StatusDependant';
+import type { FetchParams, Fetcher } from './Fetcher';
+import EventEmitter from 'events';
+
+const PROCESS = Symbol();
+
+/**
+ * A {@link Fetcher} wrapper that is {@link StatusDependant}, 
+ * enabling requests to be backlogged during inactive periods.
+ */
+export class PausableFetcher implements Fetcher, StatusDependant<boolean> {
+  protected readonly logger = getLoggerFor(this);
+  protected readonly backlog: EventEmitter[] = [];
+  protected active = false;
+  protected processing = false;
+
+  constructor(
+    protected readonly fetcher: Fetcher,
+  ) {}
+
+  async fetch(...args: FetchParams): Promise<Response> {
+    const trigger = new EventEmitter();
+    const result = new Promise<Response>(resolve => {
```

Comment:
Minor optimization, but might be nicer to already check the status here and immediately return the fetch without the promise/eventemitter if the fetcher is active.

Reply:
True, I even thought about it, but I was lazy. Will do for CSS PR ...